### PR TITLE
support multiple file analysis

### PR DESCRIPTION
**Pull request by garlicnation**
To enable iron-component-page to display docs from multiple elements.

### CODE REVIEW

File: src/analyzer.ts [hunk 1]
```diff
@@ -240,12 +244,24 @@ export class Analyzer {
...
+          content: options.content
+        }
+      ));
     }
     loader.addResolver(new NoopResolver({test: options.filter}));
 
     var analyzer = new Analyzer(false, loader);
-    return analyzer.metadataTree(href).then((root) => {
+    let additionalImports = options.additionalImports || [];
+    console.log(additionalImports);
```
Comment (garlicnation) [inline src/analyzer.ts]: Fixing this.

File: hydrolysis-analyzer.html [hunk 2]
```diff
@@ -35,6 +35,14 @@
       },
 
       /**
+       * A list of sources to load in addition to src when analyzing.
+       */
+      sources: {
```
Comment (justinfagnani) [inline hydrolysis-analyzer.html]: seems a little odd to load src + sources. Can `src` just take a `string | string[]`?

File: hydrolysis-analyzer.html [hunk 3]
```diff
@@ -83,7 +91,7 @@
      * function to kick off another analysis pass.
      */
     analyze: function() {
-      if (!this.src) {
+      if (!this.src && (!this.sources || this.sources.length == 0)) {
```
Comment (justinfagnani) [inline hydrolysis-analyzer.html]: this function is async, it should return a Promise

File: hydrolysis-analyzer.html [hunk 4]
```diff
@@ -93,14 +101,22 @@
       }
       this._setLoading(true);
 
+      var baseUri = this.ownerDocument.baseURI;
+      var additionalImports;
```
Comment (justinfagnani) [inline hydrolysis-analyzer.html]: initialize to an empty array?

Comment (justinfagnani) [inline hydrolysis-analyzer.html]: can you document `additionalImports`? The name sounds to me like additional files that are use for analysis, but maybe not part of the primary analysis targets themselves. While `sources` sounds like it's the set of entrypoint to start analyzing from. This difference in names make me look for differences in the behavior...

File: hydrolysis-analyzer.html [hunk 5]
```diff
@@ -93,14 +101,22 @@
...
         filter: this.transitive ? function() { return false; } : null,
         attachAst: true,
-        resolver: this.resolver
+        resolver: this.resolver,
+        additionalImports: additionalImports,
       };
 
-      var baseUri = this.ownerDocument.baseURI;
       var srcUrl  = new URL(this.src, baseUri).toString();
       hydrolysis.Analyzer.analyze(srcUrl, options).then(function(analyzer) {
```
Comment (justinfagnani) [inline hydrolysis-analyzer.html]: why can't `Analyzer.analyze` take a list of sources?

**Comment by Draccoz** [comment]
Any updates on that?

**Comment by rictic** [comment]
@Draccoz See `analyzePackage()` on the new `Analyzer`: https://github.com/Polymer/polymer-analyzer/blob/master/src/analyzer.ts#L75

This is a PR on the 1.0 branch (i.e. hydrolysis) and is unlikely to ever be merged.

**Pull request closed without merging**